lib/Base/Tensor.cpp

```diff
@@ -543,20 +543,68 @@ void Tensor::init(InitKind init, float val, PseudoRNG &PRNG) {
 }
 
 void Tensor::convertToType(ElemKind newTy) {
-  Tensor tmp(newTy, dims());
-  switch (newTy) {
-  case ElemKind::Float16Ty:
-    assert(getElementType() == ElemKind::FloatTy && "Cast not implemented");
-    tmp.copyWithCast<float16_t, float>(this);
-    break;
-  case ElemKind::FloatTy:
-    assert(getElementType() == ElemKind::Float16Ty && "Cast not implemented");
-    tmp.copyWithCast<float, float16_t>(this);
-    break;
-  default:
-    llvm_unreachable("Type not supported");
+  *this = this->getCopyConvertedToType(newTy);
+}
+
+Tensor Tensor::getCopyConvertedToType(ElemKind newKind) const {
+  if (!isQuantizedElemKind(newKind)) {
+    Tensor tmp(newKind, dims());
+    switch (newKind) {
+    case ElemKind::Float16Ty:
+      assert(getElementType() == ElemKind::FloatTy && "Cast not implemented");
```

Comment:
could we switch to DCHECK and add what exact cast is not implemented. Like Cast from X to Y is not implemented.